stonesoup/resampler/particle.py

```diff
@@ -35,15 +39,16 @@ def resample(self, particles):
         cdf += max_log_value
 
         # Pick random starting point
-        u_i = np.random.uniform(0, 1 / n_particles)
+        u_i = np.random.uniform(0, 1 / nparts)
 
         # Cycle through the cumulative distribution and copy the particle
         # that pushed the score over the current value
-        u_j = u_i + (1 / n_particles) * np.arange(n_particles)
+        u_j = u_i + (1 / nparts) * np.arange(nparts)
         index = weight_order[np.searchsorted(cdf, np.log(u_j))]
 
-        new_particles = particles[index]
-        new_particles.weight = np.full((n_particles, ), weight)
+        new_particles = particles[np.array(index)]
```

Comment:
Not sure why `np.array` is added here? Is there case where `np.searchsorted` returns an `int`? If so, `np.atleast_1d` may be better to be clearer.

Reply:
A leftover fix for something I removed I think. It has been changed back to `particles[index]` now.